### DISCUSSION TITLE
Decrement if data after decrement is not less than zero

### DISCUSSION THREAD

**Discussion by farellsujanto** [prisma/prisma#6465] [Q&A]
I'm looking for a method to decrement a data, if the data after the reduction is not less than zero then the data is saved to the db, else throw some error. Can anyone help me with the best method to achieve this? Thanks!

**Comment by ryands17** [comment]
@farellsujanto 👋 

You would need two calls for this. One to fetch the value and the other to update it. In this case, you can use [Optimistic Concurrency Control](https://www.prisma.io/docs/guides/prisma-guides/prisma-client-transactions-guide#optimistic-concurrency-control) to make sure that the entire operation is atomic.

**Reply by farellsujanto** [reply]
Yeah this is what I'm looking for. But my "where" is only locked to ID, i cant lookup any other variable rather than @id or @unique, can you help me with this one?

**Reply by ryands17** [reply]
You can use `updateMany` in that case as explained in the above doc.

**Reply by farellsujanto** [reply]
Thank you for your help @ryands17 , I really appreciate it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
